### PR TITLE
Redefine voice search icon placement logic

### PR DESCRIPTION
**Pull request by CrisBarreiro**
Task/Issue URL: https://app.asana.com/0/0/1205629102384066/f 

### Description
Only show voice search icon in the following cases:
* omnibar is focused and query hasn't changed
* omnibar is focused and query is empty
* DDG SERP is shown and query hasn't changed

Some additional context:
* When focusing in SERP, the omnibar text was manually being replaced with the same value to force autocomplete suggestions to show, which triggered a bunch of methods, including re-checking for voice search availability. However, this behavior forced `hasQueryChanged` to be true, which broke voice search availability checks. For this reason, a new listener has been added to trigger autocompletion rather than relying on manually setting the text again

### Steps to test this PR

**Please go to settings and enable voice search first. Requires Android 13 and device language set to English US**

_Icon shown when omnibar selected and unchanged_
- [x] Open a site, check no voice search icon is shown when not focused
- [x] Tap on the omnibar, check voice search icon is shown
- [x] Tap again on the omnibar and edit the URL
- [x] Check the voice icon isn't shown

_Icon shown when omnibar selected and query is empty
- [x] Tap on the ombibar and clear the query
- [x] Check the voice search icon is shown

_Icon shown when on SERP_ 
- [x] Search something
- [x] Check the voice search is shown even if omnibar is not focused
- [x] Tap on the omnibar, check the voice search icon is still there
- [x] Tap again on the omnibar and edit the query
- [x] Check the voice icon isn't shown

_Icon show on new tab_ 
- [x] Open a new tab
- [x] Check voice icon is there
- [x] Close keyboard
- [x] Check voice icon is still there

_Voice search icon status is reevaluated after changing setting_
- [x] Go to settings and update toggle voice search on/off
- [x] Go back and check icon gets updated without further user interaction

### Actually changed
combined
combined
combined

### Affected but unchanged
combined
combined
combined
combined
combined
combined
combined

### CODE REVIEW

**Comment by CrisBarreiro** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3477** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3477" target="_blank">Graphite</a> 
    * **PR #3585** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3585" target="_blank">Graphite</a> 
      * **PR #3587** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3587" target="_blank">Graphite</a> 
        * **PR #3607** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3607" target="_blank">Graphite</a> 
          * **PR #3608** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3608" target="_blank">Graphite</a> 
            * **PR #3613** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3613" target="_blank">Graphite</a>  👈
    * **PR #3583** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3583" target="_blank">Graphite</a> 
    * **PR #3574** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3574" target="_blank">Graphite</a> 
    * **PR #3489** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3489" target="_blank">Graphite</a> 
      * **PR #3516** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3516" target="_blank">Graphite</a> 
      * **PR #3491** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3491" target="_blank">Graphite</a> 
      * **PR #3490** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3490" target="_blank">Graphite</a> 
        * **PR #3528** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3528" target="_blank">Graphite</a> 
          * **PR #3541** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3541" target="_blank">Graphite</a> 
            * **PR #3564** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3564" target="_blank">Graphite</a> 
              * **PR #3567** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3567" target="_blank">Graphite</a> 
            * **PR #3550** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3550" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3613?utm_source=stack-comment).

**Comment by CrisBarreiro** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3477** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3477" target="_blank">Graphite</a> 
    * **PR #3585** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3585" target="_blank">Graphite</a> 
      * **PR #3587** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3587" target="_blank">Graphite</a> 
        * **PR #3607** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3607" target="_blank">Graphite</a> 
          * **PR #3608** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3608" target="_blank">Graphite</a> 
            * **PR #3613** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3613" target="_blank">Graphite</a>  👈
    * **PR #3583** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3583" target="_blank">Graphite</a> 
    * **PR #3574** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3574" target="_blank">Graphite</a> 
    * **PR #3489** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3489" target="_blank">Graphite</a> 
      * **PR #3516** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3516" target="_blank">Graphite</a> 
      * **PR #3491** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3491" target="_blank">Graphite</a> 
      * **PR #3490** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3490" target="_blank">Graphite</a> 
        * **PR #3528** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3528" target="_blank">Graphite</a> 
          * **PR #3541** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3541" target="_blank">Graphite</a> 
            * **PR #3564** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3564" target="_blank">Graphite</a> 
              * **PR #3567** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3567" target="_blank">Graphite</a> 
            * **PR #3550** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3550" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3613?utm_source=stack-comment).

File: app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt [hunk 1]
```diff
@@ -1194,22 +1194,22 @@ class BrowserTabViewModelTest {
     }
 
     @Test
-    fun whenEnteringQueryWithAutoCompleteEnabledThenAutoCompleteSuggestionsShown() {
+    fun whenTriggeringAutocompleteThenAutoCompleteSuggestionsShown() {
```
Comment (CrisBarreiro) [inline app/src/androidTest/java/com/duckduckgo/app/browser/BrowserTabViewModelTest.kt:1197]: Check PR description for an explanation on autocomplete related changes

**Review by anikiki — APPROVED** [review]
Looks good and works as expected! :tada:

**Review by anikiki — APPROVED** [review]

**Review by anikiki — APPROVED** [review]
Looks good and works as expected! :tada:

**Pull request merged**